src/superio/nuvoton/nct5104d/superio.c

```diff
@@ -106,6 +110,69 @@ static void route_pins_to_uart(struct device *dev, bool to_uart)
 	pnp_write_config(dev, 0x1c, reg);
 }
 
+static void enable_gpio_io_port(struct device *dev, bool enable_wdt1)
+{
+	u8 reg, reg1;
+	u16 io_base_address;
+	u8 uartc_enabled, uartd_enabled;
+
+	pnp_write_config(dev, 0x07, NCT5104D_GPIO_WDT);
+
+	/* 
+		Devictree always sets LDN 8 CR30.0 bit (WDT1 enable)
+		See if user really wants Watchdog Timer 1 to be enabled
+	*/ 
+
+	if(!enable_wdt1)
+	{
+		reg = pnp_read_config(dev, 0x30);
+		pnp_write_config(dev, 0x30, reg & 0xFE);
+	}
+
+	/*  
+		Check if UARTC and UARTD are both enabled
+		If they are, don't activate GPIO Address Mode
+		In any other case - activate GPIO Address Mode
+	*/
+
+	struct device *uart;
+
+	uart = dev_find_slot_pnp(SIO_PORT, NCT5104D_SP3);
+	uartc_enabled = uart->enabled;
+
+	uart = dev_find_slot_pnp(SIO_PORT, NCT5104D_SP4);
+	uartd_enabled = uart->enabled;
+
+	if (!uartc_enabled || !uartd_enabled)
+	{
+		/* 
+			Check if LDN 8 CR60 and CR61 contain valid IO Base Address
+			IO Base Address <100h ; FF8h>
+		*/
+
+		pnp_write_config(dev, 0x07, NCT5104D_GPIO_WDT);
+
+		reg = pnp_read_config(dev, 0x60);
+		reg1 = pnp_read_config(dev, 0x61);
+
+		io_base_address = (reg << 8) + reg1;
+		printk(BIOS_INFO, "SuperIO IO Base Address = %x\n", io_base_address);
+
+		if (io_base_address < 0x100 || io_base_address > 0xFF8)
+		{
+			printk(BIOS_WARNING, "SuperIO IO Base Address should be in "
```

Comment:
```suggestion
			printk(BIOS_ERROR, "Invalid IO base address %x != <100h ; FF8h> \n");
```
This should be enough. The additional info print is not needed. Too much verbosity